app/renderer/components/Inspector/Inspector.js

```diff
@@ -24,23 +29,70 @@ export default class Inspector extends Component {
     }
     this.props.bindAppium();
     this.props.applyClientMethod({methodName: 'source'});
+    this.props.getSavedActionFramework();
   }
 
   render () {
-    const {screenshot, screenshotError, selectedElement = {}, applyClientMethod, quitSession} = this.props;
+    const {screenshot, screenshotError, selectedElement = {},
+           applyClientMethod, quitSession, isRecording,
+           showRecord, recordedActions, actionFramework,
+           startRecording, pauseRecording, setActionFramework,
+           clearRecording, closeRecorder, showBoilerplate,
+           toggleShowBoilerplate} = this.props;
     const {path} = selectedElement;
 
-    let actions = <span>
-      <Button icon='arrow-left' onClick={() => applyClientMethod({methodName: 'back'})} size="small" >
-        Back
-      </Button>
-      <Button icon='reload' onClick={() => applyClientMethod({methodName: 'source'})} size="small">
-        Refresh
-      </Button>
-      <Button icon='close' onClick={() => quitSession()} size="small">
-        Quit
-      </Button>
-    </span>;
+    let actions = <ButtonGroup size="large">
+      <Tooltip title="Back">
+        <Button icon='arrow-left' onClick={() => applyClientMethod({methodName: 'back'})}/>
+      </Tooltip>
+      <Tooltip title="Refresh Source & Screenshot">
+        <Button icon='reload' onClick={() => applyClientMethod({methodName: 'source'})}/>
+      </Tooltip>
+      {!isRecording &&
+       <Tooltip title="Start Recording">
+        <Button icon="eye-o" onClick={startRecording}/>
+       </Tooltip>
+      }
+      {isRecording &&
+       <Tooltip title="Pause Recording">
+         <Button icon="pause" onClick={pauseRecording}/>
+       </Tooltip>
+      }
+      <Tooltip title="Quit Session & Close Inspector">
+        <Button icon='close' onClick={() => quitSession()}/>
+      </Tooltip>
+    </ButtonGroup>;
+
+
+    let frameworkOpts = Object.keys(frameworks).map((f) => <Option value={f}>
+      {frameworks[f].readableName}
+    </Option>);
+    let boilerplateType = showBoilerplate ? "primary" : "default";
+    let recorderActions = <div>
+      {!!recordedActions.length &&
+        <Select defaultValue={actionFramework} onChange={setActionFramework}
+         className={InspectorStyles['framework-dropdown']} size="small">
+          {frameworkOpts}
+        </Select>
+      }
+      {!!recordedActions.length &&
+        <Tooltip title="Show/Hide Boilerplate Code">
+          <Button onClick={toggleShowBoilerplate} icon="file-text"
+           size="small" type={boilerplateType}
+          />
+        </Tooltip>
+      }
+      {!!recordedActions.length &&
+        <Tooltip title="Clear Actions">
+          <Button icon="delete" onClick={clearRecording} size="small" />
+        </Tooltip>
+      }
+      {!isRecording &&
```

Comment:
These three instances of `!!recorderActions.length &&` can be replaced with one and you can either wrap the three components in a div or make them an array. Not a big deal if left as is though.